backend/models/meta.js

```diff
@@ -40,391 +40,410 @@ function clean(metadataToClean) {
-			const _branch = (model === submodel) ? branch : "master";
-			const _revId = (model === submodel) ? revId : null;
+			if (showIfcGuids) {
+				return getIFCGuids(account, submodel, shared_ids).then(ifc_guids => {
+					return {
+						account,
+						model: submodel,
+						ifc_guids
+					};
+				});
+			}
 
-			objectIdPromises.push(findModelSharedIdsByRulesQueries(
+			return {
 				account,
-				submodel,
-				positiveQueries,
-				negativeQueries,
-				_branch,
-				_revId,
-				convertSharedIDsToString && !showIfcGuids // in the case of ifcguids I need the uuid for querying and geting the ifcguids
-			).then(shared_ids => {
-				if(!shared_ids.length) {
-					return undefined;
-				}
+				model: submodel,
+				shared_ids
+			};
+		}).catch(() => {
+			// If search on a submodel failed (usually due to no revision in the submodel), it should not
+			// fail the whole API request.
+			return undefined;
+		}));
+	}
 
-				if (showIfcGuids) {
-					return getIFCGuids(account, submodel, shared_ids).then(ifc_guids => {
-						return {
-							account,
-							model: submodel,
-							ifc_guids
-						};
-					});
-				}
+	const objectIds = await Promise.all(objectIdPromises);
 
-				return {
-					account,
-					model: submodel,
-					shared_ids
-				};
-			}).catch(() => {
-				// If search on a submodel failed (usually due to no revision in the submodel), it should not
-				// fail the whole API request.
-				return undefined;
-			}));
-		}
+	return objectIds.filter((entry) => !!entry);
+};
 
-		const objectIds = await Promise.all(objectIdPromises);
+Meta.getAllIdsWithMetadataField = async (account, model, branch, rev, fieldName) => {
+	// Get the revision object to find all relevant IDs
+	let fullFieldName = "metadata";
 
-		return objectIds.filter((entry) => !!entry);
+	if (fieldName && fieldName.length > 0) {
+		fullFieldName += "." + fieldName;
 	}
 
-	async getAllIdsWithMetadataField(account, model, branch, rev, fieldName) {
-		// Get the revision object to find all relevant IDs
-		let fullFieldName = "metadata";
+	const getMeta = [];
+
+	await getSubModels(account, model, branch, rev, (subModelTS, subModel, subModelBranch, subModelRev) => {
+		getMeta.push(
+			Meta.getAllIdsWithMetadataField(subModelTS, subModel, subModelBranch, subModelRev, fieldName)
+				.then(({data}) => {
+					return {
+						data,
+						account: subModelTS,
+						model: subModel
+					};
+				})
+				.catch(() => {
+					// Just because a sub model fails doesn't mean everything failed - do nothing
+				})
+		);
+	});
 
-		if (fieldName && fieldName.length > 0) {
-			fullFieldName += "." + fieldName;
-		}
+	const subMeta = await Promise.all(getMeta);
 
-		// Check for submodel references
-		const refs = await getRefNodes(account, model, branch, rev);
+	const obj = await findNodesByField(account, model, branch, rev, fullFieldName);
 
-		const getMeta = [];
+	if (!obj) {
+		return Promise.reject(responseCodes.METADATA_NOT_FOUND);
```

Comment:
We can `throw` here?

Reply:
sure, i didn't code this (or anything else you pointed out)

All I did was remove the class 😆